### PR TITLE
Add support for NFS backend

### PR DESCRIPTION
**Pull request by nicolasbock**

### CODE REVIEW

**Comment by nicolasbock** [comment]
At this point a cherry-pick from @vuntz: https://github.com/crowbar/barclamp-cinder/compare/release/tex/master...vuntz:nfs-backend

**Comment by Itxaka** [comment]
This will collide with https://github.com/crowbar/crowbar-openstack/pull/464, at least on the migrations part.

**Comment by nicolasbock** [comment]
@Itxaka Thanks for pointing this out. I could go with revision 101, which would resolve the potential collision. I didn't see any other obvious places of collision potential, did you?

**Comment by Itxaka** [comment]
@nicolasbock I also didnt see anything that could not be merged cleanly so we should be ok in that aspect

**Comment by Itxaka** [comment]
tested locally and it works as exected

**Comment by toabctl** [comment]
+1

**Comment by Itxaka** [comment]
+1

**Comment by aplanas** [comment]
@nicolasbock is this still wip? do this needs more reivew?

**Comment by nicolasbock** [comment]
@aplanas I am currently testing it, but soon it should be good to go.

**Comment by Itxaka** [comment]
damn, I cant believe I missed that. Thats why you test it with real stuff :fire: 

Still lgtm +1

**Comment by Itxaka** [comment]
tested it thanks to @nicolasbock and seems to work as expected, creating volumes work and I can see them on the nfs server dir.
+1

**Comment by dirkmueller** [comment]
+1

File: chef/data_bags/crowbar/migrate/cinder/101_nfs_backend.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+def upgrade(ta, td, a, d)
+  a["volume_defaults"]["nfs"] = ta["volume_defaults"]["nfs"]
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/migrate/cinder/101_nfs_backend.rb]: I think we need to also check here if a["volume_defaults"]["nfs"] doesn't already exist (think about this being backported to cloud 6)

File: chef/data_bags/crowbar/migrate/cinder/101_nfs_backend.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  return a, d
+end
+
+def downgrade(ta, td, a, d)
+  a["volume_defaults"].delete("nfs")
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/migrate/cinder/101_nfs_backend.rb]: same here

**Comment by nicolasbock** [comment]
Rebased to trigger gating.

**Pull request merged**